### ISSUE TITLE
False E0050, E0185

### ISSUE THREAD

**Issue by imkow** [intellij-rust/intellij-rust#6091]
## Environment

* **IntelliJ Rust plugin version:** 0.3.131.3340-202-nightly
* **Rust toolchain version:** 1.48.0-nightly (0e2c1281e 2020-09-07) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2020.2.1 (CL-202.6948.80)
* **Operating system:** Windows 10 10.0

## Problem description
2 false errors:
Method `hello` has 2 parameters but the declaration in trait `World` has 1 [E0050] 
Method `hello` has a `self` declaration in the impl, but not in the trait [E0185]

https://github.com/google/tarpc/blob/master/example-service/src/server.rs#L27

## Steps to reproduce
```rust

#[tarpc::service]
pub trait World {
    /// Returns a greeting for name.
    async fn hello(name: String) -> String;
}

#[tarpc::server]
impl World for HelloServer {
    async fn hello(self, _: context::Context, name: String) -> String {  // Method `hello` has 2 parameters but the declaration in trait `World` has 1 [E0050] 
// Method `hello` has a `self` declaration in the impl, but not in the trait [E0185]

        format!("Hello, {}! You are connected from {:?}.", name, self.0)
    }
}

```

**Comment by Undin** [comment]
@imkow Unfortunately, proc macros are not supported yet
See #1786 for more details

**Comment by imkow** [comment]
```rust

pub trait World {
    /// Returns a greeting for name.
    fn hello(name: String) -> String;
}

impl World for HelloServer {
    fn hello(self, _: context::Context, name: String) -> String {  // Method `hello` has 2 parameters but the declaration in trait `World` has 1 [E0050] 
// Method `hello` has a `self` declaration in the impl, but not in the trait [E0185]

        format!("Hello, {}! You are connected from {:?}.", name, self.0)
    }
}
```
